### ISSUE TITLE
smoothCombine() errors out even though initialX is set to NULL

### ISSUE THREAD

**Issue by leungi** [config-i1/smooth#121]
``` r
  library(smooth)

  woiDat <-
  c(36.084, 621.21, 1214.75884615385, 1519.98571428571, 1591.66, 1691.27, 2502.48620689655, 1703.09, 1666.17, 1127.31, 1353.46, 1570.12, 1408.68, 1449.62, 1738.3, 2198.46, 2919.1, 3531.5, 4043.86, 3594.98, 2503.94, 2236.4, 2624.25, 2584.76, 1987.71, 1776.88, 1946.77, 1941.62, 1793.8, 1844.81, 1591.27, 1367.58, 1333.56, 1116.16, 905.16, 1011.18, 1032.75, 1119.53, 972.98, 996.79, 882.55, 628.62, 728.53, 1206.64, 1101.48, 808.87, 901.55, 854.653333333333, 782.874, 814.98)
  
  xReg <-
    data.frame(
      V2_steam = c(0.0975666346028447, 265.986043939106, 93.8932879271545,
                   88.0470350584574, 72.6645910864882, 120.420400130656,
                   107.430837323535, 138.630900755189, 183.428616120219, 166.645612258576,
                   170.848074510768, 173.876223541684, 195.711904886849,
                   183.018674975093, 213.251113080736, 195.97174637584, 191.364564797729,
                   198.675465383735, 230.470472846404, 303.023025577646,
                   198.740860279389, 221.621655587889, 182.583267625533, 222.24608793281,
                   252.569173882194, 275.53114820024, 268.378021462988, 277.452805658765,
                   321.066524502039, 468.473425906934, 328.165294780675,
                   276.76864891883, 278.514916242491, 311.179134856798, 347.84724678658,
                   534.98946972556, 375.791220030356, 341.339632285852,
                   338.171384239886, 307.129305580072, 298.311095908247, 367.615289466381,
                   392.076079315525, 382.216773150414, 385.313772782143,
                   370.204274651539, 382.371062663943, 327.811050361115, 253.833888573311,
                   320.299084241316),
      V3_steam = c(1.42539331130683, 205.872208318301, 248.132920550257,
                   244.87993346639, 189.097367078252, 220.013461165018,
                   225.694386493973, 321.458860968351, 406.862710030116, 410.336065315008,
                   525.82551872395, 707.896625206135, 496.216296464913,
                   725.091686476991, 675.718607247658, 555.929541984573, 612.930146718286,
                   544.302076178193, 519.705586694926, 507.665889023989,
                   576.732565612681, 497.71051590059, 522.082427655496, 476.402472063713,
                   567.07762138281, 545.655821993835, 513.156953012533, 503.443948484659,
                   476.847657285929, 574.841413317621, 514.714362684898,
                   557.747532592677, 722.875751731172, 730.331917916089, 725.621024175175,
                   709.355089121126, 727.898304251134, 706.507631211653,
                   710.875741911605, 712.988902211487, 695.752340376228, 591.260997933745,
                   609.946326547675, 642.692462521866, 641.617469192632,
                   661.74020020809, 724.604276698306, 585.357589216903, 492.277385050058,
                   559.891783180237),
      V4_liquid = c(0.0188058700226237, 270.591747171069, 294.052530201236,
                    193.978469956899, 203.088318221644, 142.996971945539,
                    132.88718644632, 116.812622255357, 142.221716342373, 219.766491380557,
                    140.779648025949, 159.795774716781, 72.0089709677361,
                    60.5418340377975, 72.0511548611398, 0.00374734078533926, 185.941895876714,
                    0.00920985810458608, 0.0127969436999403, 0.0132457484491162,
                    0.00102091885171825, 156.659920566346, 166.590694251824,
                    158.069014985655, 145.320803277241, 190.532945391582, 118.348728669863,
                    88.7025051780045, 89.2190121430252, 88.4349546146393,
                    78.365660204133, 70.4773995217588, 71.3896667858707, 176.716438339949,
                    109.037502107499, 113.032176241279, 111.199729123581,
                    45.4064018358849, 133.751338651442, 248.818625238404, 123.06503231897,
                    95.8745146204997, 103.307847048529, 131.066980491867, 161.314278942524,
                    145.150546250166, 139.979520654865, 135.644692178275,
                    114.784918269509, 54.0167899503931),
      V4_oil = c(0.0189347461611022, 243.963766796254, 262.883651024802,
                 171.370248379055, 180.412323816651, 111.790424567414,
                 108.752241234519, 95.3882266280893, 117.840356178954, 190.479232946569,
                 124.636486430708, 135.29721621641, 61.4605113655608,
                 54.0831997008715, 72.0592482402331, 0.00630104013718629, 117.827254453864,
                 0.00695915071293672, 0.00474475070834133, 0.00862145605497013,
                 0.0115124520286911, 89.8980317748319, 88.451938096229,
                 95.2575026971754, 87.3248149559647, 115.272557091434, 69.2869783923123,
                 56.2381758409273, 63.8613710113056, 63.0452930910978,
                 55.0252935690153, 43.1585693234298, 42.5648304845729, 109.699381424449,
                 67.6549379448313, 68.0122939164564, 67.1996005843021,
                 0.0081178653147067, 81.1273538056216, 123.222981016434, 62.2801156785246,
                 44.9358708615601, 84.7165371775162, 84.8866453572922,
                 75.6264878500718, 71.1523151593283, 62.1894816169981, 56.6653524771519,
                 51.3088708867009, 54.0151618223079),
      V4_steam = c(0.78742003813386, 0.521052423864603, 0.346927351877093,
                   0.394714872166514, 0.335809188336134, 0.598939095064998,
                   0.590566409006715, 0.0885772272944451, 0.277400333806872,
                   0.422080750390887, 0.0341365184634924, 0.595108260586858,
                   0.175533875450492, 0.424294931814075, 862.65359632132, 182.686197390556,
                   0.492262770235538, 0.235051270201802, 345.899645403326,
                   349.099648642912, 388.241939044893, 621.159126152441, 0.709857032075524,
                   0.308425408601761, 0.0325316023081541, 0.00958019755780692,
                   0.354088819026947, 0.544279040768743, 0.391596061736345, 0.304887778684497,
                   0.139635385945439, 0.248258195444942, 0.290646882727742,
                   0.472420822456479, 0.141427604854107, 0.0028981596231461,
                   0.693891379237175, 0.26303275860846, 0.68532646894455, 0.324089797586203,
                   0.555436084419489, 0.291134788841009, 0.261454996094108,
                   0.399006796628237, 0.018157260864973, 0.779065430909395,
                   0.370274687930942, 0.540498589724302, 0.212515977025032, 9943.40212981738),
      V4_water = c(0.00455491648986926, 26.6086488826387, 31.1612978420407,
                   22.5944229352754, 22.6883290222846, 31.1991378116421,
                   24.1217194168176, 21.4356463476177, 24.3755033636652, 29.3053117079102,
                   16.1617737893388, 24.5042083707452, 10.5376565519441,
                   6.46511103614233, 0.0034754445217549, 0.0152218041289594,
                   68.0846193745981, 0.0181994517985727, 0.0111655480042098, 0.0043704088870435,
                   0.0160668988991525, 66.7814931593146, 78.123166802004,
                   62.8185661555734, 57.9753015530296, 75.2590512195602, 49.0626130509376,
                   32.4958570394851, 25.3222257395741, 25.3615214559436,
                   23.348377024876, 27.3150596579164, 28.8047905872734, 67.0387308639567,
                   41.3692292655352, 45.0280828709994, 43.9908481742442,
                   45.4125375378504, 52.6146141218115, 125.573286808779, 60.7881554122921,
                   50.9183275041822, 18.5678972891625, 46.1723419291805, 85.67161873837,
                   73.9954166623112, 77.77330320023, 78.9806998857204,
                   63.4888751322372, 0.0187854427006092)
    )

xReg <-
data.frame(
    V2_steam = c(0.0975666346028447, 265.986043939106, 93.8932879271545,
                 88.0470350584574, 72.6645910864882, 120.420400130656,
                 107.430837323535, 138.630900755189, 183.428616120219, 166.645612258576,
                 170.848074510768, 173.876223541684, 195.711904886849,
                 183.018674975093, 213.251113080736, 195.97174637584, 191.364564797729,
                 198.675465383735, 230.470472846404, 303.023025577646,
                 198.740860279389, 221.621655587889, 182.583267625533, 222.24608793281,
                 252.569173882194, 275.53114820024, 268.378021462988, 277.452805658765,
                 321.066524502039, 468.473425906934, 328.165294780675,
                 276.76864891883, 278.514916242491, 311.179134856798, 347.84724678658,
                 534.98946972556, 375.791220030356, 341.339632285852,
                 338.171384239886, 307.129305580072, 298.311095908247, 367.615289466381,
                 392.076079315525, 382.216773150414, 385.313772782143,
                 370.204274651539, 382.371062663943, 327.811050361115, 253.833888573311,
                 320.299084241316),
    V3_steam = c(1.42539331130683, 205.872208318301, 248.132920550257,
                 244.87993346639, 189.097367078252, 220.013461165018,
                 225.694386493973, 321.458860968351, 406.862710030116, 410.336065315008,
                 525.82551872395, 707.896625206135, 496.216296464913,
                 725.091686476991, 675.718607247658, 555.929541984573, 612.930146718286,
                 544.302076178193, 519.705586694926, 507.665889023989,
                 576.732565612681, 497.71051590059, 522.082427655496, 476.402472063713,
                 567.07762138281, 545.655821993835, 513.156953012533, 503.443948484659,
                 476.847657285929, 574.841413317621, 514.714362684898,
                 557.747532592677, 722.875751731172, 730.331917916089, 725.621024175175,
                 709.355089121126, 727.898304251134, 706.507631211653,
                 710.875741911605, 712.988902211487, 695.752340376228, 591.260997933745,
                 609.946326547675, 642.692462521866, 641.617469192632,
                 661.74020020809, 724.604276698306, 585.357589216903, 492.277385050058,
                 559.891783180237),
   V4_liquid = c(0.0188058700226237, 270.591747171069, 294.052530201236,
                 193.978469956899, 203.088318221644, 142.996971945539,
                 132.88718644632, 116.812622255357, 142.221716342373, 219.766491380557,
                 140.779648025949, 159.795774716781, 72.0089709677361,
                 60.5418340377975, 72.0511548611398, 0.00374734078533926, 185.941895876714,
                 0.00920985810458608, 0.0127969436999403, 0.0132457484491162,
                 0.00102091885171825, 156.659920566346, 166.590694251824,
                 158.069014985655, 145.320803277241, 190.532945391582, 118.348728669863,
                 88.7025051780045, 89.2190121430252, 88.4349546146393,
                 78.365660204133, 70.4773995217588, 71.3896667858707, 176.716438339949,
                 109.037502107499, 113.032176241279, 111.199729123581,
                 45.4064018358849, 133.751338651442, 248.818625238404, 123.06503231897,
                 95.8745146204997, 103.307847048529, 131.066980491867, 161.314278942524,
                 145.150546250166, 139.979520654865, 135.644692178275,
                 114.784918269509, 54.0167899503931),
      V4_oil = c(0.0189347461611022, 243.963766796254, 262.883651024802,
                 171.370248379055, 180.412323816651, 111.790424567414,
                 108.752241234519, 95.3882266280893, 117.840356178954, 190.479232946569,
                 124.636486430708, 135.29721621641, 61.4605113655608,
                 54.0831997008715, 72.0592482402331, 0.00630104013718629, 117.827254453864,
                 0.00695915071293672, 0.00474475070834133, 0.00862145605497013,
                 0.0115124520286911, 89.8980317748319, 88.451938096229,
                 95.2575026971754, 87.3248149559647, 115.272557091434, 69.2869783923123,
                 56.2381758409273, 63.8613710113056, 63.0452930910978,
                 55.0252935690153, 43.1585693234298, 42.5648304845729, 109.699381424449,
                 67.6549379448313, 68.0122939164564, 67.1996005843021,
                 0.0081178653147067, 81.1273538056216, 123.222981016434, 62.2801156785246,
                 44.9358708615601, 84.7165371775162, 84.8866453572922,
                 75.6264878500718, 71.1523151593283, 62.1894816169981, 56.6653524771519,
                 51.3088708867009, 54.0151618223079),
    V4_steam = c(0.78742003813386, 0.521052423864603, 0.346927351877093,
                 0.394714872166514, 0.335809188336134, 0.598939095064998,
                 0.590566409006715, 0.0885772272944451, 0.277400333806872,
                 0.422080750390887, 0.0341365184634924, 0.595108260586858,
                 0.175533875450492, 0.424294931814075, 862.65359632132, 182.686197390556,
                 0.492262770235538, 0.235051270201802, 345.899645403326,
                 349.099648642912, 388.241939044893, 621.159126152441, 0.709857032075524,
                 0.308425408601761, 0.0325316023081541, 0.00958019755780692,
                 0.354088819026947, 0.544279040768743, 0.391596061736345, 0.304887778684497,
                 0.139635385945439, 0.248258195444942, 0.290646882727742,
                 0.472420822456479, 0.141427604854107, 0.0028981596231461,
                 0.693891379237175, 0.26303275860846, 0.68532646894455, 0.324089797586203,
                 0.555436084419489, 0.291134788841009, 0.261454996094108,
                 0.399006796628237, 0.018157260864973, 0.779065430909395,
                 0.370274687930942, 0.540498589724302, 0.212515977025032, 9943.40212981738),
    V4_water = c(0.00455491648986926, 26.6086488826387, 31.1612978420407,
                 22.5944229352754, 22.6883290222846, 31.1991378116421,
                 24.1217194168176, 21.4356463476177, 24.3755033636652, 29.3053117079102,
                 16.1617737893388, 24.5042083707452, 10.5376565519441,
                 6.46511103614233, 0.0034754445217549, 0.0152218041289594,
                 68.0846193745981, 0.0181994517985727, 0.0111655480042098, 0.0043704088870435,
                 0.0160668988991525, 66.7814931593146, 78.123166802004,
                 62.8185661555734, 57.9753015530296, 75.2590512195602, 49.0626130509376,
                 32.4958570394851, 25.3222257395741, 25.3615214559436,
                 23.348377024876, 27.3150596579164, 28.8047905872734, 67.0387308639567,
                 41.3692292655352, 45.0280828709994, 43.9908481742442,
                 45.4125375378504, 52.6146141218115, 125.573286808779, 60.7881554122921,
                 50.9183275041822, 18.5678972891625, 46.1723419291805, 85.67161873837,
                 73.9954166623112, 77.77330320023, 78.9806998857204,
                 63.4888751322372, 0.0187854427006092))

# es() runs fine   
es(woiDat, model="ZZZ", h=12, holdout=TRUE, xreg=xReg, xregDo="select",
  updateX=TRUE, silent=TRUE, persistenceX = NULL, transitionX = NULL)

# smoothCombine() errors out even though initialX is set to NULL (estimated), so
# code should've generated the correct size, based on input to xreg
smoothCombine(woiDat, h=6, holdout=TRUE, xreg=xReg, xregDo="select",
             updateX=TRUE, silent=FALSE, persistenceX = NULL, transitionX = NULL,
             initialX = NULL)

#> Error: The size of initial vector for exogenous is wrong!
#> It should correspond to the number of exogenous variables.
```

Created on 2018-06-30 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by config-i1** [comment]
Hi Leungi,

Thanks for this! I'll fix this as soon as I have time (a week maybe?).

**Comment by config-i1** [comment]
This is now solved in cd83914fbb5a583e1593273baf6fc4b0e33f9a76 Update from github for this or wait for a smooth v2.4.5 on CRAN

**Comment by leungi** [comment]
@config-i1, thanks for the quick update!
